Data struct reconfig

birdday opened the pull request:
Made changes to the analysis of Henry's Coefficients to work with pandas dataframes/csv files rather than messier dictionary format equivalents. Also added tests directory with unit tests for these new functions.

paulboone commented on tests/__init__.py:
```diff
@@ -0,0 +1 @@
+
```
Usually this file would be empty. The whitespace is unecessary. I'd create it via "touch tests/__init__.py". 

Do you have the whitespace package enabled in atom? (are you using atom?)

birdday replied:
Just double-checked, this file is empty (i.e. no whitespace). I'm guessing github just shows the above since the file was created. And yes, I am using the whitespace add-on.

paulboone commented on sensor_array/helper/henrys_coefficient_analysis.py:
```diff
@@ -32,27 +33,9 @@ def yaml_loader(filepath):
-        return keys, reader_list
+def import_simulated_data(filename):
+    ads_data = pd.read_csv(filename, sep='\t', engine='python')
+    return ads_data
```
This doesn't look like an atomic commit since before you were returning keys + reader_list and now your are returning just the results of the pandas read_csv. Don't the calling functions need to be modified too?

paulboone commented on sensor_array/helper/henrys_coefficient_analysis.py:
```diff
@@ -38,258 +38,120 @@ def import_simulated_data(filename):
+    r2, rmse, ybar = calculate_r2_and_rmse(p, comps[0:i], mass[0:i])
 
-    return a[0], max_comp, R2, RMSE, len(comps)-i
+    return p, r2, rmse
```
again, doesn't this break the calling functions?

paulboone commented on sensor_array/helper/henrys_coefficient_analysis.py:
```diff
@@ -1,32 +1,7 @@
-import time
-import tensorflow as tf
-import tensorflow_probability as tfp
-
```
ideally, these should have been removed when you removed the code

birdday replied:
Can you elaborate on your comment? (Or just tell me if we do a call, mostly writing this comment as a reminder to ask.)

paulboone commented on tests/tests_henrys_coefficient_analysis.py:
```diff
@@ -0,0 +1,76 @@
+    weight_type='error'
+    fixed_intercept=False
+
+    p, max_comp, r2, rmse, i =  hca.calculate_kH(ads_data, gas, eval_type=eval_type, r2_min=r2_min, rmse_min=rmse_min, weight_type=weight_type, fixed_intercept=fixed_intercept)
```
I would inline the arguments here, r2_min=0.90, so you don't have all the r2_min=r2_min terms. _Maybe_ it adds clarity for the positional args, but not for the keyword args.

birdday replied:
Another comment I'm a bit confused about. That is a parameter I want to be able to set in the config file, but I also wanted a default value for it. If you think it should not have a default value, then I could change it.

paulboone commented on tests/tests_henrys_coefficient_analysis.py:
```diff
@@ -0,0 +1,76 @@
+
+
+def test_create_weight_matrix():
+    error = [0,1,2,4,5]
```
Your tests are an opportunity to define in as human-readable a way as possible what the heck your code should be doing. So just like a good commit message isn't "added unit tests", a good test name isn't going to be "test_*thismethod*". It should typically be a description of the behavior, or roughly a description of what SHOULD happen. So possibly a good name for the first assert case could be: 
"test_weight_matrix_with_no_weighting_is_evenly_distributed" 

That said, you don't want to spend TOO much time coming up with the perfect names; the important part is that people reviewing the test can tell (A) what the test failing means, and (b) what should happen so they can see if the test is written correctly.

Read this: https://enterprisecraftsmanship.com/posts/you-naming-tests-wrong/

I'd like you to think about this and possible ways to rename them to be clearer. It's ok if we don't make any serious changes here though; it is already so much better having the test coverage.

birdday committed:
Removed unused read_kH_results function.

This function was unused as of the last commit which removed the plotting of all Henry's coefficients.

birdday committed:
Removed extra junk code.

Removed a small section of code that was not meant to be commited, but was and additionally made it through the previous sloppy PR.

birdday committed:
Removed all instances of plotting.

Removed plotting of adsorption data and subseuqent fit from analyze_all_ratios function.
Removed the summary plot of all Henry's coefficients from the end of 'execute_henrys_coefficient_analysis.py'.

The latter call was actually broken to begin with, as the file containing the plotting function was never imported. This commit is atomic by itself, and makes the previous commit atomic, but per the reason in the previous commit message, was done as a separate commit.

birdday committed:
Removed previous analysis for individual and multiple ratios, as this is not consistent with a to-be-implemented data structure.

N.B. This commit is atomic in that all remaining functions perfrom as intended, it just intentionally removes all code from the execute file.

birdday committed:
Removed unused imports / unnecessary comments. The unused imports should have been removed in the previous PR, but got through as it was a rushed and sloppy PR, hence the appearence of a non-atomic commit. I double checked the files though, and this commit is in fact atomic.

birdday committed:
Removed plotting functions and moved subset to a separate file.

N.B. This commit is NOT atomic, as there are still instances in which these functions are called in the scripts 'henrys_coefficient_analysis.py' and 'execute_henrys_coefficient_analysis.py', however these are all removed in the susequent commit.

The reason for doing this as two separate commits is that the one call in 'execute_henrys_coefficient_analysis.py' was actually broken to begin with, as the function was never imported. The breaking of this file occured as part of the previous PR, but was not caught.

birdday committed:
Updated import function to use new data structure (pandas dataframes).

N.B. All analysis functions are updated in the subsequent commit.

birdday committed:
Updated all analysis functions to work with new data structures.

birdday committed:
Absorbed execute file into main file, and rewrote it to be used with a YAML config file.

birdday committed:
Updated .gitignore to enable tracking of sample config and data files and added sample data and config files for henrys_coefficient_analysis.

paulboone reviewed (approved):
LGTM

birdday commented:
LGTM from Paul (We already covered the commit together, no need to re-review when merging to WilmerLab).
